tx-pool/src/pool.rs

```diff
@@ -263,7 +263,12 @@ impl TxPool {
     }
 
     // Remove transactions from the pool until total size <= size_limit.
-    pub(crate) fn limit_size(&mut self, callbacks: &Callbacks) {
+    pub(crate) fn limit_size(
+        &mut self,
+        callbacks: &Callbacks,
+        current_entry_id: Option<&ProposalShortId>,
+    ) -> Result<(), Reject> {
```

Comment:
Given that `current_entry_id` is an `Option`, should the return type be `Option<Reject>` too?

The primary function of `limit_size` is to remove transactions to ensure the total size fits within the size_limit, so its execution should always be successful and not return an `Err`.

Reply:
make sense, then we use `map_or` to convert it into `Err` if necessary.